actions/signing-event/action.yml

```diff
@@ -77,3 +112,17 @@ runs:
           })
 
           await core.summary.addHeading(summary).write()
+
+    - name: Dispatch another signing event workflow
```

Comment:
Just curiosity, what is the reason the workflow is kicked off again?

Reply:
The issue is different behaviour for custom and default token: since the workflow is triggered by push, the workflow may automatically "self-trigger" (if it does make a new commit for the targets metadata changes) when using a custom token _but not when using the default token_... The options that I see for solving this are:
* Avoid push triggers completely somehow (I don't think this is practical)
* Let the signing-event sometimes run twice but exit early if we think we already ran once for the same commit
* Make sure the signing-event always runs twice after targets metadata changes

I've chosen the 3rd option because it actually fits the process. There are now two possible paths when the signing event is triggered:
1. targets metadata changes are needed: only do those changes, then let signing-event trigger again (either automatically or by manual dispatch)
1. targets metadata changes are not needed: output signing event status

Reply:
I tried to document this at the top of the action but as you can see from the previous comment it's I've not exactly figured how to explain the logic succinctly...

Reply:
One part that is unclear to me is:

```
#     * When using a custom token this triggers push event handler
#     * When using the default GitHub token the action calls createWorkflowDispatch()
```

Why is there a different action taken with a custom token vs the default token? I would have thought that by providing a custom token, you're simply overriding the default and the same path is taken.

Reply:
It's how GitHub works:

> When you use the repository's GITHUB_TOKEN to perform tasks, events triggered by the GITHUB_TOKEN, with the exception of workflow_dispatch and repository_dispatch, will not create a new workflow run.

That could be more explicit, what it's not saying out loud is that _events triggered by other tokens will create new workflow runs_.

Reply:
Ah, thanks, didn’t realize that restriction exists. I guess it’s to prevent using up available quota? Nonetheless, makes sense why this is set up this way.